db/maintenance/info_test.go

```diff
@@ -0,0 +1,75 @@
+package maintenance
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/Scalingo/cli/utils"
+	"github.com/Scalingo/go-scalingo/v6"
+)
+
+func TestGetNextMaintenanceLocalWindow(t *testing.T) {
+	now := dateOfTheWeekUTC(time.Wednesday, 8)
+
+	tests := map[string]struct {
+		maintenanceTimeData        time.Time
+		expectedMaintenanceRunTime func(time.Time, scalingo.MaintenanceStatus) time.Time
+	}{
+		"when the maintenance should be executed next week": {
+			maintenanceTimeData: dateOfTheWeekUTC(time.Sunday, 0),
+			expectedMaintenanceRunTime: func(maintenanceData time.Time, _ scalingo.MaintenanceStatus) time.Time {
+				return maintenanceData.AddDate(0, 0, 7)
+			},
+		},
+		"when the maintenance window is ongoing": {
+			maintenanceTimeData: dateOfTheWeekUTC(time.Wednesday, 7),
+			expectedMaintenanceRunTime: func(maintenanceData time.Time, maintenanceType scalingo.MaintenanceStatus) time.Time {
+				if maintenanceType == scalingo.MaintenanceStatusScheduled {
```

Comment:
**question (non-blocking):** do you really need to put conditions on your test setup? it makes the test a bit harder to read and understand